src/MessageHandler/SearchNoHitsMessageHandler.php

```diff
@@ -45,151 +45,312 @@ public function __construct(
         private readonly LoggerInterface $logger,
         private readonly SearchService $searchService,
         private readonly VendorImageValidatorService $validatorService,
-        private readonly MetricsService $metricsService
+        private readonly MetricsService $metricsService,
+        private readonly iterable $singleIdentifierVendors
     ) {
     }
 
     /**
+     * Invoke handler.
+     *
      * @param SearchNoHitsMessage $message
+     */
+    public function __invoke(SearchNoHitsMessage $message): void
+    {
+        try {
+            $identifier = $message->getIdentifier();
+            $found = false;
+
```

Comment:
Not required (just for fun): could these if statements have been written as match statement? Not sure it will help the code og make it more readble.

Reply:
Not sure `match` is a good match (pun intended). `match`is used for assignment based on known variables. Here we have one primary variable `$found` that is modified through the execution flow.